x-pack/plugins/apm/public/components/app/ServiceOverview/__test__/NoServicesMessage.test.tsx

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { shallow } from 'enzyme';
+import React from 'react';
+import { NoServicesMessage } from '../NoServicesMessage';
+
+describe('No Services Message', () => {
+  it('should render correctly when historical data is found', () => {
+    const wrapper = shallow(<NoServicesMessage historicalDataFound={true} />);
+    expect(wrapper).toMatchSnapshot();
```

Comment:
This is fine for now, but I think it would be great with a discussion about how we use snapshot.
I think we over-use them a bit. In this case, if the output changes the slightest the test will fail - which causes noise, and it's easy to overlook if something significant actually changed.

Reply:
I think we probably need to write up a clear guide on how we want to do testing and what kinds of testing we want to do in what situations, so that we can refer to it as we develop.